src/screens/FeedScreen/elements/Search.js

```diff
@@ -6,35 +6,47 @@ import StringConstant from '../../../utils/string/StringConstant';
 import {fonts} from '../../../utils/fonts';
 import {COLORS, FONTS, SIZES} from '../../../utils/theme';
 
-const Search = ({animatedValue, onContainerClicked = () => {}, getSearchLayout}) => {
+const Search = ({
+  animatedValue,
+  onContainerClicked = () => {},
+  getSearchLayout,
+  children,
+  containerStyle
+}) => {
   const onSearchLayout = (event) => {
     const {height} = event.nativeEvent.layout;
     if (getSearchLayout) {
       getSearchLayout(height);
     }
   };
 
+  if (children) {
+    return (
+      <Animated.View
+        onLayout={onSearchLayout}
+        style={[HeaderStyles.animatedViewContainer(animatedValue), containerStyle]}>
+        <View style={[HeaderStyles.wrapperSearch, {backgroundColor: 'white'}]}>{children}</View>
+      </Animated.View>
+    );
+  }
+
   return (
-    <Animated.View onLayout={onSearchLayout} style={styles.animatedViewContainer(animatedValue)}>
-      <Pressable onPress={onContainerClicked} style={styles.searchContainer}>
-        <View style={styles.wrapperSearch}>
-          <View style={styles.wrapperIcon}>
+    <Animated.View
+      onLayout={onSearchLayout}
+      style={[HeaderStyles.animatedViewContainer(animatedValue), containerStyle]}>
+      <Pressable onPress={onContainerClicked} style={HeaderStyles.searchContainer}>
+        <View style={HeaderStyles.wrapperSearch}>
+          <View style={HeaderStyles.wrapperIcon}>
             <MemoIc_search width={16.67} height={16.67} />
           </View>
-          <Text style={styles.inputText}>{StringConstant.newsTabHeaderPlaceholder}</Text>
+          <Text style={HeaderStyles.inputText}>{StringConstant.newsTabHeaderPlaceholder}</Text>
         </View>
       </Pressable>
     </Animated.View>
   );
 };
 
-const styles = StyleSheet.create({
-  container: {
-    flexDirection: 'row',
-    backgroundColor: 'white',
-    marginBottom: SIZES.base,
-    marginHorizontal: SIZES.base
-  },
+export const HeaderStyles = StyleSheet.create({
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a conditional rendering based on the presence of `children` prop. This is a good practice for making components more reusable. However, there's a hardcoded color value in line 28 which should be replaced with a constant from the theme file.
```diff
- <View style={[HeaderStyles.wrapperSearch, {backgroundColor: 'white'}]}>{children}</View>
+ <View style={[HeaderStyles.wrapperSearch, {backgroundColor: COLORS.white}]}>{children}</View>
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->